### ISSUE TITLE
issue with python3.7?

### ISSUE THREAD

**Issue by truatpasteurdotfr** [simeoncarstens/pandemic-sim#1]
Hi Simeon,

I tried to use directly your repository on mybinder (https://mybinder.org/v2/gh/simeoncarstens/pandemic-sim/57044e22492d3159340e79c2bc35329ccc590136) and the simulation is failing with:
```
jovyan@jupyter-simeoncarstens-2dpandemic-2dsim-2dev2g4yr3:~$ python3 simulation.py
Simulating step 0/300...
Simulating step 50/300...
Simulating step 100/300...
Simulating step 150/300...
Simulating step 200/300...
Simulating step 250/300...
Animating step 0/300...
Animating step 50/300...
Animating step 100/300...
Animating step 150/300...
Animating step 200/300...
Animating step 250/300...
Animating step 300/300...
Traceback (most recent call last):
  File "simulation.py", line 393, in <module>
    anim.save("output.mp4", fps=10)
  File "/srv/conda/envs/notebook/lib/python3.7/site-packages/matplotlib/animation.py", line 1102, in save
    alt_writer = next(writers, None)
TypeError: 'MovieWriterRegistry' object is not an iterator
```
Python3.7 VS 3.6 ? I assume that you are using 3.6 from the nix file :P.

cheers

Tru

**Comment by simeoncarstens** [comment]
Hi Tru!

Nice to hear from you and thanks for opening this issue!
Seems like either

- a bug in (at least the most recent version of) `matplotlib` (https://github.com/matplotlib/matplotlib/issues/17097) or `celluloid` (https://github.com/jwkvam/celluloid/issues/13) 
- and / or the the fact that `ffmpeg` is not available in the binder Docker image. It was also missing in the `shell.nix`, where I added it now. I now also mention the requirements in the readme.

Locally, I have no issues with Python 3.7 (nor 3.6, obviously) and `matplotlib` 3.2.1. I tested it with 3.2.0, which works well, too. It fails on binder with 3.2.0, though. With 3.1.3, running it in binder, it complains about `ffmpeg` not being available.
In a pure Nix shell without `ffmpeg` it gives a different, more informative error, correctly complaining that `ffmpeg` is missing.

I'm not really sure what to do about this&mdash;I'll leave this open for now; I might just change to some other animation solution, which would close this issue. It would be nicest not to rely on `ffmpeg` or any additional Python dependency such as `celluloid`.
In any case, I now preemptively pin the `matplotlib` version to 3.2.0, just to be sure. 

Thanks also for introducing me to binder&mdash;I didn't know about it!
If you're interested in reproducible Jupyter notebooks, you might be interested in [jupyterWith](https://github.com/tweag/jupyterWith), which provides reproducible Jupyter environments via Nix.
